avocado/utils/pmem.py

```diff
@@ -319,3 +319,109 @@ def destroy_namespace(self, namespace='all', region='', bus='',
                           shell=True, ignore_status=True):
             raise PMemException('Namespace destroy command failed')
         return True
+
+    def write_infoblock(self, namespace='', stdout=False, output=None,
+                        **kwargs):
+        """
+        Write an infoblock to the specified medium, either of namespace, stdout
+        and output which are mutually exclusive to each other.
+
+        :param namespace: Write the infoblock to given namespace
+        :param stdout: Write the infoblock to stdout if True
+        :param output: Write the infoblock to the file path specified
+        :param **kwargs:
+        Example:
+           pmem.write_infoblock(namespace=ns_name, align=align,
+                                size=size, mode='devdax')
+
+        :return: True if command succeeds
+        :rtype: bool
+        :raise: :class:`PMemException`, if command fails.
+        """
+        if not (namespace or stdout or output):
+            raise PMemException("Specify atleast one output medium")
+
+        args_dict = {'region': '-r %s', 'bus': '-b %s', 'mode': '-m %s',
+                     'memmap': '-M %s', 'size': '-s %s', 'align': '-a %s',
+                     'uuid': '-u %s', 'parent_uuid': '-p %', 'offset': '-O %s'}
+
+        if namespace:
+            args = namespace
+        elif stdout:
+            args = "-c"
+        elif output:
+            args = "-o %s" % output
+
+        if namespace == "all":
+            if 'region' in kwargs and kwargs['region']:
+                args += " %s" % args_dict[key] % kwargs.pop('region')
+            if 'bus' in kwargs and kwargs['bus']:
+                args += " %s" % args_dict[key] % kwargs.pop('bus')
```

Comment:
This block of code is very similar of what we have on the second method. Could you try to split this and reuse to avoid duplication lines? Maybe we could reduce the complexity here that it is still too high.